symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/parsing/MatchResult.java

```diff
@@ -0,0 +1,38 @@
+package com.symphony.bdk.core.activity.parsing;
+
+import lombok.Getter;
+import org.apiguardian.api.API;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Class representing the outcome of a matching between a {@link SlashCommandPattern} and a {@link com.symphony.bdk.gen.api.model.V4Message}.
+ * It can contain the map of arguments if applicable. Key is argument name, value is the actual value in the message.
+ * Argument value can be of type {@link String}, {@link Mention}, {@link Cashtag} or {@link Hashtag}.
+ */
+@API(status = API.Status.INTERNAL)
+@Getter
+public class MatchResult {
+
+  private boolean isMatching;
+  private Map<String, Object> arguments;
```

Comment:
would we have an interest in having a custom object here with:
- get(property)
- getMention(property)
- getCashtag(property)
- ....

to save the user from having to cast?

Reply:
Also having our own object would give us immutability